hawkular-bus/hawkular-bus-common/src/main/java/org/hawkular/bus/common/BinaryData.java

```diff
@@ -59,68 +59,52 @@ public void setOnCloseAction(Runnable action) {
         onCloseAction = action;
     }
 
-    public int read() throws IOException {
-        if (unreadInMemoryDataExists()) {
-            return (inMemoryData[inMemoryDataPointer++] & 0xff);
-        } else {
-            return streamData.read();
-        }
-    }
+    public void close() throws IOException {
+        backingStream.close();
```

Comment:
Yeah, that's a good catch. We should have been doing that before. We should add it now.